sql/src/main/java/io/crate/expression/scalar/string/EncodeDecodeFunction.java

```diff
@@ -0,0 +1,126 @@
+ * agreement.
+ */
+
+package io.crate.expression.scalar.string;
+
+import io.crate.common.Hex;
+import io.crate.common.Octal;
+import io.crate.expression.scalar.ScalarFunctionModule;
+import io.crate.expression.scalar.arithmetic.BinaryScalar;
+import io.crate.metadata.FunctionInfo;
+import io.crate.types.DataTypes;
+
+import java.nio.charset.StandardCharsets;
+import java.util.Base64;
+import java.util.Locale;
+import java.util.function.BinaryOperator;
+
+public class EncodeDecodeFunction {
+
+    public static void register(ScalarFunctionModule module) {
+        module.register(new BinaryScalar<>(new Encode(), "encode", DataTypes.STRING, FunctionInfo.DETERMINISTIC_ONLY));
+        module.register(new BinaryScalar<>(new Decode(), "decode", DataTypes.STRING, FunctionInfo.DETERMINISTIC_ONLY));
+    }
+
+    /**
+     * Takes a binary data of "bytea" type and encodes it into the given output format.
+     */
+    private static class Encode implements BinaryOperator<String> {
+
+        @Override
+        public String apply(String bytea, String format) {
+            switch (Format.valueOf(format.toUpperCase(Locale.ROOT))) {
+                case BASE64:
+                    final byte[] text;
+                    if (Hex.isHexFormat(bytea)) {
+                        text = Hex.decodeHex(Hex.stripHexFormatFlag(bytea));
+                    } else {
+                        text = Octal.decode(bytea);
+                    }
+                    return Base64.getEncoder().encodeToString(text);
+                case HEX:
+                    if (Hex.isHexFormat(bytea)) {
+                        // the input is already in hex format
+                        return Hex.validateHex(Hex.stripHexFormatFlag(bytea));
+                    }
+                    return Hex.encodeHexString(Octal.decode(bytea));
+                case ESCAPE:
+                    if (Hex.isHexFormat(bytea)) {
+                        return Octal.encode(Hex.decodeHex(Hex.stripHexFormatFlag(bytea)));
+                    }
+                    return Octal.encode(bytea.getBytes(StandardCharsets.UTF_8));
+                default:
+                    // should not happen
+                    throw new RuntimeException();
+            }
+        }
+
+    }
+
+    /**
+     * Takes a string encoded into the given format and returns its hexadecimal representation.
+     */
+    private static class Decode implements BinaryOperator<String> {
+
+        @Override
+        public String apply(String text, String format) {
+            final byte[] value;
+
+            switch (Format.valueOf(format.toUpperCase(Locale.ROOT))) {
+                case BASE64:
+                    value = Base64.getDecoder().decode(text.getBytes(StandardCharsets.UTF_8));
+                    break;
+                case HEX:
+                    // text is already in hex format
+                    return String.format(Locale.ROOT, "%s%s", Hex.HEX_FLAG, Hex.validateHex(text));
```

Comment:
I think using string concatenation is faster and cheaper here than using the string format.